src/views/forms/AccountManagementForms/SourceSelectMenu.tsx

```diff
@@ -85,10 +85,14 @@ export const SourceSelectMenu = ({
         </Styled.Text>
       ),
     }))
-    .filter(
-      (chain) =>
-        type === TransferType.deposit || !!cctpTokensByChainId[chain.value] || !CCTPWithdrawalOnly
-    )
+    .filter((chain) => {
+      if (type === TransferType.deposit) {
+        // if deposit, return only CCTP tokens UNLESS CCTPDepositOnly is not enabled
+        return !!cctpTokensByChainId[chain.value] || !CCTPDepositOnly;
```

Comment:
[note] we can condense this all into a 1 liner but i felt like expanding this and adding comments helps make this more clear and extendable for any future FF's or features